src/qinfer/resamplers.py

```diff
@@ -61,7 +62,7 @@
 
 class Resampler(with_metaclass(ABCMeta, object)):
     @abstractmethod
-    def __call__(self,  model, particle_weights, particle_locations,
+    def __call__(self,  model, particle_dist,
```

Comment:
I can't think of any better way to do this, and like the new API you're suggesting here, but it is also a breaking change. That said, I can't think of any third-party code with custom resamplers that couldn't be readily fixed. At any rate, following semver, we should make sure to label the new version at least 1.1 as a result.